firmware/lib/Axis/src/Axis.cpp

```diff
@@ -0,0 +1,317 @@
+#include "Axis.h"
+#include "Debug.h"
+#include "Timer.h"
+
+// -------------------------------- X AXIS SETUP -------------------------------- //
+#define ACCEL_X                 100
+#define VEL_MIN_X               100
+#define VEL_MAX_X               1000
+#define STEP_PIN_X              5 // need to know the register for fast operation pin 5 - PC25 
+#define DIR_PIN_X               6 // need to know the register for fast operation pin 6 - PC24
+#define ENCODER_PIN_A_X         7
+#define ENCODER_PIN_B_X         8
+#define LIMIT_SW_HOME_PIN_X     9
+#define LIMIT_SW_FAR_PIN_X      10
+
+AxisConfig axis_x_config = {DIR_PIN_X, STEP_PIN_X, ENCODER_PIN_A_X, ENCODER_PIN_B_X, LIMIT_SW_HOME_PIN_X, LIMIT_SW_FAR_PIN_X,
+                            ACCEL_X, VEL_MAX_X, VEL_MIN_X, TC1, 0, 1, TC3_IRQn, TC4_IRQn, isr_encoder_x, isr_limit_switch_x};
+
+Axis setup_axis(AxisConfig *axis_config)
```

Comment:
I guess I missed this on the first round, but generally speaking you don't want to return large structs from functions because they're being passed around by value. The standard practice would be to do something like this:
```c
void setup_axis(AxisConfig *axis_config, Axis *axis)
{
    axis->blah = blah;
    axis->blah2 = blah2;
    ...
}

AxisConfig axis_x_config = ...;
Axis axis_x;

void some_init_method()
{
    setup_axis(&axis_x_config, &axis_x);
}
```